src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -61,62 +59,49 @@ internal abstract partial class AbstractInitializeParameterCodeRefactoringProvid
                 return;
             }
 
-            var containingMember = parameterNode.FirstAncestorOrSelf<TMemberDeclarationSyntax>();
-            if (containingMember == null)
+            var functionDeclaration = parameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
+            if (functionDeclaration == null)
             {
                 return;
             }
 
             var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
             var parameterDefault = syntaxFacts.GetDefaultOfParameter(parameterNode);
-
+          
             // Don't offer inside the "=initializer" of a parameter
             if (parameterDefault?.Span.Contains(position) == true)
             {
                 return;
             }
-
+         
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var method = (IMethodSymbol)semanticModel.GetDeclaredSymbol(containingMember, cancellationToken);
-            if (method.IsAbstract ||
-                method.IsExtern ||
-                method.PartialImplementationPart != null ||
-                method.ContainingType.TypeKind == TypeKind.Interface)
-            {
-                return;
-            }
 
+            // we can't just call GetDeclaredSymbol on functionDeclaration because it could an anonymous function,
+            // so first we have to get the parameter symbol and then its containing method symbol
             var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
-            if (!method.Parameters.Contains(parameter))
+            if (parameter == null || parameter.Name == "")
             {
                 return;
             }
 
-            if (parameter.Name == "")
+            var method = (IMethodSymbol)parameter.ContainingSymbol;
+            if (method.IsAbstract ||
+                method.IsExtern ||
+                method.PartialImplementationPart != null ||
+                method.ContainingType.TypeKind == TypeKind.Interface)
             {
                 return;
             }
 
-            // Only offered on method-like things that have a body (i.e. non-interface/non-abstract).
-            var bodyOpt = GetBody(containingMember);
-
             // We support initializing parameters, even when the containing member doesn't have a
             // body. This is useful for when the user is typing a new constructor and hasn't written
             // the body yet.
-            var blockStatementOpt = default(IBlockOperation);
-            if (bodyOpt != null)
-            {
-                blockStatementOpt = semanticModel.GetOperation(bodyOpt, cancellationToken) as IBlockOperation;
-                if (blockStatementOpt == null)
-                {
-                    return;
-                }
-            }
-
+            var blockStatementOpt = GetBlockOperation(functionDeclaration, semanticModel, cancellationToken);
```

Comment:
i would still like the null check for this.

Reply:
do what if it's null? it can be null as mentioned in the comment

Reply:
ah, i see.  ok then :)